src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
+const MAX_HEALTH = 100;
+const BITE_DAMAGE = 50;
+const DEAD_VALUE = 0;
+
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name) {
+    this.name = name;
+    this.health = MAX_HEALTH;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (
+      animal instanceof Herbivore &&
+      !(animal instanceof Carnivore) &&
```

Comment:
Checking that the victim is not a predator is unnecessary here.
Since the condition animal instanceof Herbivore already guarantees that the victim is not a predator.
```suggestion
```

Reply:
It was an idea from AI 😄
![image](https://github.com/mate-academy/js_herbivores_and_carnivores/assets/21338158/94489bc2-325e-411d-a9cc-95fac1df1b81)